src/otk/utils.py

```diff
@@ -0,0 +1,102 @@
+import json
+from typing import TypeVar, Generic
+
+
+def _deep_convert(data):
+    """Recursively convert nested dicts and lists into HiddenAttrDict and HiddenAttrList."""
+    if isinstance(data, (HiddenAttrDict, HiddenAttrList)):
```

Comment:
This is maybe a silly question - but why do we need this? The yaml parser will build the nesting for us AFAICT (i.e. everywhere we need the HiddenAttr* clases we have them via yaml constructions) - if I (naviely) remove this helper it, nothing in the tests break (except the one test that is designed to test this). So what am I missing?

Reply:
This was actually ment for places in the code just working with dict and not caring about the types. e.g. initializing the class with a full dict (I thought this will be needed, but I could also stash it somewhere and pull out once really needed (other than tests which I thought make sense to make the class robust)